odoo/osv/expression.py

```diff
@@ -1297,9 +1319,6 @@ def to_sql(self):
 
         assert len(stack) == 1
         query = stack[0]
-        joins = ' AND '.join(self.joins)
-        if joins:
-            query = '(%s) AND %s' % (joins, query)
 
         params.reverse()
         return query, params
```

Comment:
maybe the `expression` object could return a `Query` object with everything in it (tables, joins, where clause and params)